src/CancelController.ts

```diff
@@ -1,23 +1,22 @@
-import {CancelTokenSource} from 'axios'
-const {CancelToken} = require('axios')
+export interface CancelControllerInterface {
+  cancel(): void;
+  onCancel(fn: Function): void;
+}
 
-class CancelController {
-  private axiosCancel: CancelTokenSource
+class CancelController implements CancelControllerInterface {
+  private promise: Promise<void>
+  private resolve: (value?: void | PromiseLike<void> | undefined) => void = () => {}
```

Comment:
We don't need value here

Reply:
Because:

![image](https://user-images.githubusercontent.com/1747220/70605934-78cb5c00-1c0c-11ea-9ab7-69abc79a8476.png)